### PR TITLE
[Dart] Make clients crossplatform

### PR DESCRIPTION
**Pull request by nickmeinhold**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
  - @ircecho @swipesight @jaumard 

### Description of the PR

Since 12.0 the http package has had conditional imports so creating a Client() will use the appropriate option of either a BrowserClient or an IOClient depending on the platform so there would seem to be no need for the generator to use a browserClient flag to distinguish between Flutter and the web.  

I didn't create an issue as there was already #3540 (although it's been closed so I wasn't sure what I should do).

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR. I've some suggestions. Please PM me in [Slack](https://join.slack.com/t/openapi-generator/shared_invite/enQtNzAyNDMyOTU0OTE1LTY5ZDBiNDI5NzI5ZjQ1Y2E5OWVjMjZkYzY1ZGM2MWQ4YWFjMzcyNDY5MGI4NjQxNDBiMTlmZTc5NjY2ZTQ5MGM) when you've time.

File: modules/openapi-generator/src/main/resources/dart2/pubspec.mustache [hunk 1]
```diff
@@ -4,6 +4,6 @@ description: {{pubDescription}}
 environment:
   sdk: '>=2.0.0 <3.0.0'
 dependencies:
-  http: '>=0.11.1 <0.13.0'
+  http: '>=0.12.0 <0.13.0'
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/dart2/pubspec.mustache]: @nickmeinhold To make the change backward compatible, what about revert the changes you've made and only make the following change?

`  http: '>={{^browserClient}}0.12.0{{/browserClient}}{{#browserClient}}0.11.1{{/browserClient}} <0.13.0'`

Comment (nickmeinhold) [reply modules/openapi-generator/src/main/resources/dart2/pubspec.mustache]: Yeah absolutely, if you think that's the way to go.  My thinking was that because 0.12.0 was where the http package started using conditional imports, it was important to bump the dependency, as the change to [api_cient](modules/openapi-generator/src/main/resources/dart2/api_client.mustache) assumes conditional imports.  Does that make sense?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/dart2/pubspec.mustache]: Right. I just want this change to be backward compatible so that it can be merged into master right away. Otherwise, the change should be targeting 4.2.0 branch, which will be released 2 months later.

Comment (nickmeinhold) [reply modules/openapi-generator/src/main/resources/dart2/pubspec.mustache]: OK cool, I'll make the change.

Comment (nickmeinhold) [reply modules/openapi-generator/src/main/resources/dart2/pubspec.mustache]: Oh shoot, I think I misunderstood - you're suggesting I revert all the other changes so the whole PR is just a single change of that one line, is that right?

**Comment by wing328** [comment]
I'll update the wording in the CLI option in another PR.

**Comment by wing328** [comment]
@nickmeinhold thanks for the PR, which has been included in the v4.1.1 release: https://twitter.com/oas_generator/status/1165944867391860737
